### PR TITLE
E2C: Show snapshot error status

### PR DESCRIPTION
**Pull request by joshhunt**
Shows an error alert banner when the snapshot migration is in the ERROR state. It's pretty tricky to trigger this locally - I've been manually editing my database to get it into this state.

 - Creates a new `getError()` function that takes all the possible API calls and returns an error message to display in the UI.
 - Shows a new soft error state for when the migration has completed but some resources failed.
 - Suppresses the error toast on all requests in the RTKQ slice
 - Stops polling if we were unable to load the snapshot

7954_2024-07-30-17-29_firefox

### CODE REVIEW

File: public/app/features/migrate-to-cloud/onprem/Page.tsx [hunk 1]
```diff
@@ -251,3 +241,107 @@ export const Page = () => {
...
+      title: t('migrate-to-cloud.onprem.snapshot-error-status-title', 'Error migrating resources'),
+      body: t(
+        'migrate-to-cloud.onprem.snapshot-error-status-body',
+        'There was an error creating the snapshot or starting the migration process. See the Grafana server logs for more details'
+      ),
+    };
+  }
+
+  const errorCount = snapshot?.stats?.statuses?.['ERROR'] ?? 0;
+  if (errorCount > 0) {
```
Comment (joshhunt) [inline public/app/features/migrate-to-cloud/onprem/Page.tsx:336]: this should check for when the migration has completed

File: public/app/features/migrate-to-cloud/api/index.ts [hunk 2]
```diff
@@ -8,50 +8,39 @@ export const cloudMigrationAPI = generatedAPI.enhanceEndpoints({
 
   endpoints: {
     // Cloud-side - create token
-    getCloudMigrationToken(endpoint) {
```
Comment (joshhunt) [inline public/app/features/migrate-to-cloud/api/index.ts:11]: There's lots of changes in this file, but all I did was remove `suppressErrorsOnQuery` and rearrange them better.

File: public/app/features/migrate-to-cloud/onprem/Page.tsx [hunk 3]
```diff
@@ -251,3 +244,107 @@ export const Page = () => {
...
+  createSnapshotError: unknown; // From createSnapshotMutation
+  uploadSnapshotError: unknown; // From uploadSnapshotMutation
+  cancelSnapshotError: unknown; // From cancelSnapshotMutation
+  disconnectSnapshotError: unknown; // From disconnectMutation
+}
+
+interface ErrorDescription {
+  title: string;
+  body: string;
+  severity?: 'error' | 'warning';
```
Comment (ashharrison90) [inline public/app/features/migrate-to-cloud/onprem/Page.tsx:261]: my 2c: `severity` should be required or fallback to `error` on line 202

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

**Review by mmandrus — APPROVED** [review]
Nice touch 🪄 LGTM

**Pull request merged**